### PR TITLE
[chore] implement unit tests for cosmos sdk 0.50 bump

### PR DESCRIPTION
**Pull request by colmazia**
Fixed: #XXXX

## Please ensure the following requirements are met before submitting a pull request:

- [ ] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [ ] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [ ] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: app/modules.go [hunk 1]
```diff
@@ -237,7 +236,7 @@ func simulationModules(
 		),
 		ibc.NewAppModule(app.IBCKeeper),
 		app.TransferModule,
-		app.ICAModule,
+		// app.ICAModule, // remove ICA simulation as we don't have ICAController
```
[the remark this answers is not in the document]
Comment (RogerKSI) [reply app/modules.go:239]: removed

File: app/keepers/keepers.go [hunk 2]
```diff
@@ -390,8 +390,7 @@ func NewAppKeeper(
 		appCodec,
 		appKeepers.keys[ibctransfertypes.StoreKey],
 		appKeepers.GetSubspace(ibctransfertypes.ModuleName),
-		// TODO: Check this field
-		appKeepers.IBCFeeKeeper, // ISC4 Wrapper: PFM Router middleware
+		appKeepers.IBCFeeKeeper, // ICS4 Wrapper: PFM Router middleware
```
[the remark this answers is not in the document]
Comment (RogerKSI) [reply app/keepers/keepers.go:393]: removed

File: x/oracle/keeper/validator_status_test.go [hunk 3]
```diff
@@ -58,7 +61,7 @@ func (suite *KeeperTestSuite) TestAllocateTokenNoActiveValidators() {
 
 	// No active oracle validators so nothing should happen.
 	// Expect only try to sum of validator power
-	suite.mockValidators()
+	oracletestutil.ChainGoMockCalls(suite.mockValidators()...)
```
[the remark this answers is not in the document]
Comment (colmazia) [reply x/oracle/keeper/validator_status_test.go]: fixed

File: yoda/executor/executor_test.go [hunk 4]
```diff
@@ -8,36 +8,36 @@ import (
...
-	require.EqualError(t, err, "invalid executor, cannot parse executor: beeb")
+	_, _, _, err := parseExecutor("test")
+	require.EqualError(t, err, "invalid executor, cannot parse executor: test")
 }
 
 func TestParseExecutorInvalidTimeoutError(t *testing.T) {
-	_, _, _, err := parseExecutor("beeb:www.beebprotocol.com?timeout=beeb")
-	require.EqualError(t, err, "invalid timeout, cannot parse duration with error: time: invalid duration \"beeb\"")
+	_, _, _, err := parseExecutor("test:www.testprotocol.com?timeout=test")
+	require.EqualError(t, err, "invalid timeout, cannot parse duration with error: time: invalid duration \"test\"")
```
Comment (RogerKSI) [inline yoda/executor/executor_test.go:42]: replace all testprotocol to bandprotocol instead.

**Review by RogerKSI — CHANGES_REQUESTED** [review]

File: x/oracle/keeper/request_test.go [hunk 5]
```diff
@@ -1,262 +1,277 @@
...
-// ) {
-// 	if resolveStatus == types.RESOLVE_STATUS_OPEN {
-// 		require.False(t, k.HasResult(ctx, rid))
-// 	} else {
-// 		r, err := k.GetResult(ctx, rid)
-// 		require.NoError(t, err)
-// 		require.NotNil(t, r)
-// 		require.Equal(t, resolveStatus, r.ResolveStatus)
-// 	}
+func testRequest(
```
[the remark this answers is not in the document]
Comment (colmazia) [reply x/oracle/keeper/request_test.go]: done

File: yoda/executor/executor_test.go (hunk 4 above)
Comment (colmazia) [reply yoda/executor/executor_test.go]: fixed

File: testing/test_helpers.go [hunk 6]
```diff
@@ -255,15 +256,15 @@ func GenesisStateWithValSet(app *band.BandApp, dir string) band.GenesisState {
...
-	oracleGenesis.DataSources = generateDataSources(dir)
-	oracleGenesis.OracleScripts = generateOracleScripts(dir)
+	oracleGenesis.DataSources = GenerateDataSources(dir)
+	oracleGenesis.OracleScripts = GenerateOracleScripts(dir)
 	genesisState[oracletypes.ModuleName] = app.AppCodec().MustMarshalJSON(oracleGenesis)
 
 	return genesisState
 }
 
 // generateDataSources generates a set of data sources for the BandApp.
```
Comment (taobun) [inline testing/test_helpers.go:266]: comment

**Review by RogerKSI — APPROVED** [review]

**Pull request merged**